src/components/Reports/GoalCalculator/GoalCard/GoalCard.tsx

```diff
@@ -65,113 +73,168 @@ const StyledActionBox = styled(Box)({
 });
 
 export interface GoalCardProps {
-  goalId: number;
-  goalTitle: string;
-  goalAmount: number;
-  goalDate: DateTime;
-  starred: boolean;
-  onStarToggle: (goalId: number) => void;
-  onDelete: (goalId: number) => void;
-  onView: (goalId: number) => void;
+  goal: ListGoalCalculationFragment;
+
+  /** Remove this prop and always render the star once we do something with the primary flag*/
+  renderStar?: boolean;
 }
 
 export const GoalCard: React.FC<GoalCardProps> = ({
-  goalId,
-  goalTitle,
-  goalAmount,
-  goalDate,
-  starred,
-  onStarToggle,
-  onDelete,
-  onView,
+  goal,
+  renderStar = false,
 }) => {
   const { t } = useTranslation();
   const locale = useLocale();
+  const accountListId = useAccountListId() ?? '';
+  const [updateGoalCalculation] = useUpdateGoalCalculationMutation();
+  const [deleteGoalCalculation] = useDeleteGoalCalculationMutation();
+  const [deleting, setDeleting] = useState(false);
+
+  const handleStarClick = async () => {
+    await updateGoalCalculation({
+      variables: {
+        accountListId,
+        attributes: {
+          id: goal.id,
+          primary: !goal.primary,
+        },
+      },
+      optimisticResponse: {
+        updateGoalCalculation: {
+          goalCalculation: {
+            ...goal,
+            primary: !goal.primary,
+          },
+        },
+      },
+      refetchQueries: ['GoalCalculations'],
+    });
+  };
 
-  const handleStarClick = () => {
-    onStarToggle(goalId);
+  const handleDeleteClick = () => {
+    setDeleting(true);
   };
 
-  const handleDelete = () => {
-    onDelete(goalId);
+  const handleConfirmDelete = async () => {
+    await deleteGoalCalculation({
+      variables: {
+        accountListId,
+        id: goal.id,
+      },
+      update: (cache) => {
+        cache.evict({ id: `GoalCalculation:${goal.id}` });
+        cache.gc();
+      },
+    });
+    setDeleting(false);
```

Comment:
Lets add snackbar to this so the user is updated if it's successful or there is an error

Reply:
Quoting my other comment:

> The GraphQL client automatically shows a snackbar message if there was an error. I know in some places we show a snackbar message on errors, but that actually results in 2 snackbar messages popping up.

I'm not strongly opposed to adding a success message. Although my initial thought is that seeing the goal calculation disappear from the list is sufficient to tell the user that it was deleted.